microraiden/microraiden/client/client.py

```diff
@@ -73,34 +67,32 @@ def __init__(
         if not web3:
             if channel_manager_proxy:
                 self.web3 = channel_manager_proxy.web3
+                self.channel_manager_address = channel_manager_proxy.address
             elif token_proxy:
                 self.web3 = token_proxy.web3
             else:
-                if not rpc:
```

Comment:
Won't this be problematic? What if you want to use ethereum node that's not on the localhost (or Infura or something)?

Reply:
Then pass a `Web3(RPCProvider('somewhere.else', 8545))` as `web3`.

This doesn't really add any complexity but simplifies the ctor signature quite a bit.

Reply:
@pcppcp , can this be closed?

Reply:
ah, I see. I am also in favor of having ctor use less params.